crates/swc_ecma_transforms_react/src/jsx/mod.rs

```diff
@@ -596,13 +640,52 @@ where
 
                 self.top_level_node = top_level_node;
 
+                let args = once(name.as_arg()).chain(once(props_obj.as_arg()));
+                let args = if self.development {
+                    // set undefined literal to key if key is None
+                    let key = match key {
+                        Some(key) => key,
+                        None => ExprOrSpread {
+                            spread: None,
+                            expr: Box::new(build_undefined_literal_value()),
+                        },
+                    };
+
+                    // set undefined literal to __source if __source is None
+                    let __source = match __source {
+                        Some(__source) => __source,
+                        None => ExprOrSpread {
+                            spread: None,
+                            expr: Box::new(build_undefined_literal_value()),
+                        },
+                    };
+
+                    // set undefined literal to __self if __self is None
+                    let __self = match __self {
+                        Some(__self) => __self,
+                        None => ExprOrSpread {
+                            spread: None,
+                            expr: Box::new(build_undefined_literal_value()),
+                        },
+                    };
+                    args.chain(Some(key))
+                        .chain(once(
+                            Lit::Bool(Bool {
+                                span: DUMMY_SP,
+                                value: use_jsxs,
+                            })
+                            .as_arg(),
+                        ))
+                        .chain(Some(__source))
```

Comment:
There's `std::iter::once` and it's better than using `Some` to create an iterator.

Reply:
👍   75a12b8